src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_kds.c

```diff
@@ -190,7 +190,14 @@ static int xocl_add_context(struct xocl_dev *xdev, struct kds_client *client,
 
 	mutex_lock(&client->lock);
 	/* If this client has no opened context, lock bitstream */
-	if (!client->num_ctx) {
+	if (!client->ctx) {
```

Comment:
Hi @saifuddin-xilinx 

I am trying to understand the logic here. With multiple xclbin a single KDS client can support multiple "kds_client_ctx" but seems with your logic here that you will only "allocate" kds_client_ctx once i.e very first time when client->ctx is null. I fail to understand how will you allocate more kds_client_ctx for multiple xclbins?

Reply:
Got it. Thanks!